crates/dojo-bindgen/src/utils/torii_typegen/types.rs

```diff
@@ -31,12 +29,13 @@ pub trait TypeHandler {
             Token::Tuple(tuple) => self.get_tuple(tuple),
             Token::Composite(composite) => self.get_composite(composite),
             Token::GenericArg(arg) => self.get_generic_arg(arg),
-            t => panic!("Unsupported type {t:#?}!")
+            t => panic!("Unsupported type {t:#?}!"),
         }
     }
 
     fn get_basic_type(&self, type_name: &str) -> Type {
         let ccsp = self.get_cairo_serde_path();
+        let type_name = type_name.split("::").last().unwrap_or(type_name);
```

Comment:
I may have to check, but this may cause some issues when using derived types (like `Iter<Item = &str>`, that may be accessed using `T::Item` in some situations.

I may need to think a bit about how we could fix it.

This may also be worth putting into its own function when we do combine declarations.

Reply:
Yeah, this is just a temporary solution I've found for now, since I'm not fully sure how the `Type` type works yet, but if you do find something we can make the appropriate changes.

Reply:
Sounds good, let's merge it in the meantime though